website/site/content/docs/collection-types.md

```diff
@@ -70,37 +71,38 @@ A `files` collection contains one or more uniquely configured files. Unlike item
 
 When configuring a `files` collection, each file in the collection is configured separately, and listed under the `files` field of the collection. Each file has its own list of `fields`, and a unique filepath specified in the `file` field (relative to the base of the repo).
 
-**Note:** Files listed in a file collection must already exist in the repo, and must have a valid value for the file type. For example, an empty file works as valid YAML, but a JSON file must have a non-empty value to be valid, such as an empty object.
+**Note:** When NetlifyCMS connects to your backend, it fetches files on your default branch (typically `master`). For this reason, files listed in a file collection must already exist in the repo. Files must also have a valid value for the file type. For example, an empty file works as valid YAML, but a JSON file must have a non-empty value to be valid, such as an empty object.
```

Comment:
Default branch selection could vary by backend in the future, there's no central handling for it. Currently we don't check for any default branch, we just use master if you haven't explicitly set a branch in your config.yml. That considered it may be best to leave this change out.

Reply:
Thanks for your review!

I agree with your point about the default branch being a configurable detail, I can reword that. However, to me this is the most important change in the PR since it addresses the issue I was facing: not fully understanding that NetlifyCMS reads files directly from the configured Git repository, and not hosted files (locally or remote). I feel this needs to be written somewhere explicitly.

As a compromise, could one of the following ideas work? Either:

1) I could add a sentence to the [Accessing the CMS](https://www.netlifycms.org/docs/add-to-your-site/#accessing-the-cms) paragraph: 
>Note: when editing content, NetlifyCMS fetches files on your configured git repository branch. If you are running NetlifyCMS locally with changed files, or on a hosted site with outdated files, content within the admin UI may not reflect hosted files.
2) I could modify the sentence in this PR:
>When NetlifyCMS connects to your backend, it fetches files on your [configured repository branch](https://www.netlifycms.org/docs/add-to-your-site/#backend).

Reply:
Ah, understood. I think both changes would be worth adding. Our docs definitely need to a better job of explaining core concepts.